pandas/tests/frame/test_axis_select_reindex.py

```diff
@@ -117,6 +117,16 @@ def test_drop(self):
         df.drop(labels=df[df.b > 0].index, inplace=True)
         assert_frame_equal(df, expected)
 
+    def test_drop_empty_index(self):
+        # GH 27994
+        df_index = pd.DataFrame(
+            {"a": [0, 1]},
+            index=[pd.Timestamp("2019-01-01"), pd.Timestamp("2019-01-01")],
+        )
+        assert_frame_equal(df_index.drop([]), df_index)
+        assert_frame_equal(df_index.drop(np.array([])), df_index)
+        assert_frame_equal(df_index.drop(df_index[:0]), df_index)
+
```

Comment:
Can you add asserts for things like

```
.drop('a')
.drop(['a'])
```

Reply:
@TomAugspurger Ended up creating another test for those in a new commit while fixing the `TypeError` issue. Can you check?